runtime/common/src/lib.rs

```diff
@@ -512,3 +512,51 @@ pub mod changes {
 		}
 	}
 }
+
+/// Module for investment portfolio common to all runtimes
+pub mod investment_portfolios {
+
+	use cfg_traits::{InvestmentsPortfolio, TrancheCurrency};
+	use sp_std::vec::Vec;
+
+	/// Get the PoolId, CurrencyId, InvestmentId, and Balance for all
+	/// investments for an account.
+	pub fn get_portfolios<
+		Runtime,
+		AccountId,
+		TrancheId,
+		Investments,
+		InvestmentId,
+		CurrencyId,
+		PoolId,
+		Balance,
```

Comment:
Living this type on runtime, maybe passing pallet_investment::Config, alleviate its usage.

Reply:
From my experimenting with that there seems to be a bit of a bound constraint hell getting all the trait type parameters in investments_pallet::Config, especially with getting the AssetId from the Tokens type the pallets CurrencyOf is using, which is used by all the trait impls.  I can take a look later though with fresher eyes, I might just be missing something since it's late.

Reply:
If you need more Config bounds than `pallet_investment::Config`, then will be bounds collisions, and then it could be a bit hellish, yes